packages/salesforcedx-vscode-soql/src/editor/soqlEditorProvider.ts

```diff
@@ -40,35 +45,32 @@ export class SOQLEditorProvider implements vscode.CustomTextEditorProvider {
         )
       ]
     };
-
     webviewPanel.webview.html = this.getWebViewContent(webviewPanel.webview);
-  }
+    const instance = new SOQLEditorInstance(document, webviewPanel, _token);
+    this.instances.push(instance);
+    instance.onDispose(this.disposeInstance.bind(this));
 
+    Array.prototype.push.apply(
+      this.context.subscriptions,
+      instance.subscriptions
+    );
+  }
   private getWebViewContent(webview: vscode.Webview): string {
     const pathToLwcDist = path.join(
       this.context.extensionPath,
       SOQL_BUILDER_UI_PATH
     );
-    const pathToHtml = path.join(pathToLwcDist, 'index.html');
+    const pathToHtml = path.join(pathToLwcDist, HTML_FILE);
     let html = fs.readFileSync(pathToHtml).toString();
-    const scriptUri = webview.asWebviewUri(
-      vscode.Uri.file(path.join(pathToLwcDist, 'app.js'))
-    );
-    const zeroDotScriptUri = webview.asWebviewUri(
-      vscode.Uri.file(path.join(pathToLwcDist, '0.app.js'))
-    );
-    const cspMetaTag = `<meta
-      http-equiv="Content-Security-Policy"
-      content="default-src 'none';
-      img-src ${webview.cspSource} https:;
-      script-src ${webview.cspSource};
-      style-src 'unsafe-inline' ${webview.cspSource};"
-    />`;
-
-    html = html.replace('<!-- CSP TAG -->', cspMetaTag);
-    html = html.replace('./0.app.js', `${zeroDotScriptUri}`);
-    html = html.replace('./app.js', `${scriptUri}`);
-
+    html = EditorUtils.transformHtml(html, pathToLwcDist, webview);
```

Comment:
Is there any way this could fail? If it were in a try catch is there a way we could gracefully fail? I wonder if we want some static Error webview of default page like a custom 404 error page.

Reply:
It would be disasterous, but i guess a bad dist directory could would be un-recoverable.     Let's keep it in mind, but won't implement it here.